homeassistant/components/simplisafe/__init__.py

```diff
@@ -283,30 +299,155 @@ def v3_only(coro):
     return True
 
 
+@dataclass(frozen=True)
+class SimpliSafeWebsocketEvent:
+    """Define a representation of a parsed websocket event."""
+
+    event_data: dict
+
+    changed_by: Optional[str] = field(init=False)
+    event_type: Optional[str] = field(init=False)
+    info: str = field(init=False)
+    sensor_name: str = field(init=False)
+    sensor_type: EntityTypes = field(init=False)
+    timestamp: datetime = field(init=False)
+
+    def __post_init__(self):
+        """Initialize."""
+        object.__setattr__(self, "changed_by", self.event_data["pinName"])
+        object.__setattr__(
+            self, "event_type", get_event_type_from_payload(self.event_data)
+        )
+        object.__setattr__(self, "info", self.event_data["info"])
+        object.__setattr__(self, "sensor_name", self.event_data["sensorName"])
+        object.__setattr__(
+            self, "timestamp", utc_from_timestamp(self.event_data["eventTimestamp"])
+        )
+
+        try:
+            object.__setattr__(
+                self, "sensor_type", EntityTypes(self.event_data["sensorType"]).name
+            )
+        except ValueError:
+            _LOGGER.warning(
+                'Encountered unknown entity type: %s ("%s"). Please report it at'
+                "https://github.com/home-assistant/home-assistant/issues.",
+                self.event_data["sensorType"],
+                self.event_data["sensorName"],
+            )
+            object.__setattr__(self, "sensor_type", None)
+
+
+class SimpliSafeWebsocket:
+    """Define a SimpliSafe websocket "manager" object."""
+
+    def __init__(self, hass, websocket):
+        """Initialize."""
+        self._hass = hass
+        self._websocket = websocket
+        self._websocket_reconnect_delay = DEFAULT_SOCKET_MIN_RETRY
+        self._websocket_watchdog_listener = None
+        self.last_events = {}
+
+        hass.loop.create_task(self.async_websocket_connect())
```

Comment:
From Python 3.7 we can use `asyncio.create_task`.

Reply:
We don't want side effects in init method. Make a setup method or just create this task after creating the instance.